src/test/java/com/qaprosoft/carina/demo/mfp/MobileTest.java

```diff
@@ -71,4 +73,76 @@ public void caloriesValidationOnQuickAddTest() {
         Assert.assertEquals(quickAddPage.getCalories(), 17, "Calories calculation is wrong");
     }
 
+    @Test
+    @MethodOwner(owner = "Hostiuk")
+    @TestRailCases(testCasesId = "4")
+    @TestLabel(name = "feature", value = {"mobile", "regression"})
+    public void customDashboardTest() {
+        LoginPageBase loginPage = initPage(getDriver(), LoginPageBase.class);
+        loginPage.login(R.TESTDATA.get("email"), R.TESTDATA.get("password"));
+
+        CommonPageBase commonPage = initPage(getDriver(), CommonPageBase.class);
+        DiaryPageBase diaryPage = (DiaryPageBase) commonPage.clickBottomNavigatorButton(BottomNavigatorButtons.DIARY);
+        diaryPage.selectCustomDashboard(CustomDashboardPageBase.Options.CALORIES_FOCUS);
+        SoftAssert softAssert = new SoftAssert();
+        softAssert.assertTrue(commonPage.isItemByTextPresent(IConstants.GOAL),
+                IConstants.GOAL + " label is not present");
+        softAssert.assertTrue(commonPage.isItemByTextPresent(IConstants.FOOD),
+                IConstants.FOOD + " label is not present");
+        softAssert.assertTrue(commonPage.isItemByTextPresent(IConstants.EXERCISE),
+                IConstants.EXERCISE + " label is not present");
+        softAssert.assertTrue(commonPage.isItemByTextPresent(IConstants.REMAINING),
+                IConstants.REMAINING + " label is not present");
+        softAssert.assertAll();
+
+        diaryPage.selectCustomDashboard(CustomDashboardPageBase.Options.MACRONUTRIENT_FOCUS);
+        softAssert = new SoftAssert();
+        softAssert.assertTrue(commonPage.isItemByTextPresent(IConstants.CARBS),
+                IConstants.CARBS + " label is not present");
+        softAssert.assertTrue(commonPage.isItemByTextPresent(IConstants.FAT),
+                IConstants.FAT + " label is not present");
+        softAssert.assertTrue(commonPage.isItemByTextPresent(IConstants.PROTEIN),
+                IConstants.PROTEIN + " label is not present");
+        softAssert.assertTrue(commonPage.isItemByTextPresent(IConstants.CALORIES),
+                IConstants.CALORIES + " label is not present");
+        softAssert.assertAll();
```

Comment:
you just use once softAssert.assertAll() in the end of the test and once create new SoftAssert in the very beginning of the test. it has no sense cuz you never observe a test run in real time - you just see the result.